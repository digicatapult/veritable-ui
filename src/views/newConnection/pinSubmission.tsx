import Html from '@kitajs/html'
import { singleton } from 'tsyringe'
import { UUID, pinCodeRegex } from '../../models/strings.js'
import { Page } from '../common.js'
import { NewConnectionTemplates } from './base.js'

@singleton()
export class PinSubmissionTemplates extends NewConnectionTemplates {
  constructor() {
    super()
  }

  public renderPinForm = (props: {
    connectionId: UUID
    pin?: string
    continuationFromInvite: boolean
    remainingTries?: string
  }) => {
    const stepCount = props.continuationFromInvite ? 3 : 2
    const remainingTriesString = props.remainingTries ? props.remainingTries : ''
    return (
      <Page
        title="Veritable - New Connection"
        activePage="connections"
        heading="New Connection - PIN Verification"
        headerLinks={[
          { name: 'Connections', url: '/connection' },
          { name: 'Pin Submission', url: `/connection/${props.connectionId}/pin-submission` },
        ]}
        stylesheets={['connection.css']}
      >
        <div class="connections header">
          <span>PIN Code submission</span>
        </div>
        <this.newConnectionForm
          submitRoute={`${props.connectionId}/pin-submission`}
          progressStep={stepCount - 1}
          progressStepCount={stepCount}
          actions={[
            { type: 'link', text: 'Fill In Later', href: '/connection' },
            { type: 'submit', value: 'submitPinCode', text: 'Continue' },
          ]}
        >
          <div class="accented-container">
            <p>Please enter the verification code from the physical letter</p>
            <input name="stepCount" value={stepCount.toString()} type="hidden" />
            <input
              id="new-connection-invite-input-pin"
              name="pin"
              placeholder="Code"
              required
              value={props.pin || ''}
              type="text"
              pattern={pinCodeRegex.source}
              oninput="this.reportValidity()"
              minlength={6}
              maxlength={6}
            />
            <p class="warn-pin-attempts">{Html.escapeHtml(remainingTriesString)}</p>
            <img class="htmx-indicator" src="/public/images/send.svg" />
          </div>
        </this.newConnectionForm>
      </Page>
    )
  }

  public renderSuccess = (props: { companyName: string; stepCount: number }): JSX.Element => {
    return (
      <this.newConnectionForm
        feedback={{
          type: 'message',
          message: `PIN code has been sucessfully submitted and will need to be verified by the issuer now.`,
        }}
        progressStep={props.stepCount}
        progressStepCount={props.stepCount}
        actions={[{ type: 'link', text: 'Back To Home', href: '/connection' }]}
      >
        <div id="new-connection-invite-input">
          <p>
            PIN Code has been submitted for
            {Html.escapeHtml(props.companyName)} company ID. Please wait for the verification code to be confirmed by
            viewing theverification status.
          </p>
        </div>
      </this.newConnectionForm>
    )
  }
  public renderError = (props: { companyName: string; stepCount: number; errorMessage: string }): JSX.Element => {
    return (
      <this.newConnectionForm
        progressStep={props.stepCount}
        progressStepCount={props.stepCount}
        actions={[{ type: 'link', text: 'Back To Home', href: '/connection' }]}
      >
        <div id="new-connection-invite-input">
          <p class="warn-pin-attempts">
            You have run out of PIN attempts for
            {Html.escapeHtml(props.companyName)} company ID.
            {Html.escapeHtml(props.errorMessage)}
          </p>
        </div>
      </this.newConnectionForm>
    )
  }
}
