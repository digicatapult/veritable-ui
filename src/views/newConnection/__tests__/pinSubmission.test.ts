import { expect } from 'chai'
import { describe, it } from 'mocha'

import { PinSubmissionTemplates } from '../pinSubmission.js'

describe('PinSubmissionTemplates', () => {
  describe('show form', () => {
    it('should render form as a continuation of from invite', async () => {
      const templates = new PinSubmissionTemplates()
      const rendered = await templates.renderPinForm({ connectionId: 'CONNECTION_ID', continuationFromInvite: true })
      expect(rendered).to.matchSnapshot()
    })

    it('should render form as a stand alone flow', async () => {
      const templates = new PinSubmissionTemplates()
      const rendered = await templates.renderPinForm({ connectionId: 'CONNECTION_ID', continuationFromInvite: false })
      expect(rendered).to.matchSnapshot()
    })

    it('should render form with PIN', async () => {
      const templates = new PinSubmissionTemplates()
      const rendered = await templates.renderPinForm({
        connectionId: 'CONNECTION_ID',
        continuationFromInvite: true,
        pin: '123456',
      })
      expect(rendered).to.matchSnapshot()
    })
  })
})
