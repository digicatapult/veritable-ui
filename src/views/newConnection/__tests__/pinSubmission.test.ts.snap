// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewInviteTemplates show form should render a web page with the a form in an empty state 1`] = `"<form id=\\"new-connection-form\\" hx-post=\\"/connection/new/pin-submission\\" hx-select=\\"#new-connection-form > *\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 1\\"><div id=\\"new-connection-progress-text\\"><span>Step 2</span><span> of 2</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div id=\\"new-connection-invite-input\\"><p>PIN Code 123456 has been submitted for 00000001 company ID. Please wait for the verification code to be confirmed by viewing the verification. status.</p></div><div id=\\"new-connection-feedback\\" class=\\"accented-container feedback-neutral\\">[submitPinCode]: PIN code has been sucessfully submitted and will need to be verified by the issuer now.</div><div id=\\"new-connection-actions\\"><a class=\\"button   button-filled\\" href=\\"/connection\\"><span class=\\"button-text \\">Back To Home</span></a></div></form>"`;

exports[`NewInviteTemplates show form should render form with a error message and invalid response 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/new-invite.css\\"/><title>Veritable - New Connection</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon \\" style=\\"--background-image: url(&#34;/public/images/category.svg&#34;)\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"--background-image: url(&#34;/public/images/connection.svg&#34;)\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/folder.svg&#34;)\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/notification.svg&#34;)\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/setting.svg&#34;)\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">New Connection - PIN Verification</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><span class=\\"url-separator\\">/</span><a title=\\"Pin Submission\\" href=\\"/connection/new/00000001/pin-submission\\">Pin Submission</a></div></div><div class=\\"content main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><div class=\\"card-body\\"><form id=\\"new-connection-form\\" hx-post=\\"/connection/new/pin-submission\\" hx-select=\\"#new-connection-form > *\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.5\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 2</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"connectionId\\" value=\\"00000001\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" class=\\"new-connection-input-field\\" placeholder=\\"Code\\" required value=\\"\\" type=\\"text\\" pattern=\\"^[1-9][0-9]{5}$\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button   \\" href=\\"/connection\\"><span class=\\"button-text \\">Fill In Later</span></a><button class=\\"button   button-filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span class=\\"button-text \\">Continue</span></button></div></form></div></div></main></body></html>"`;

exports[`NewInviteTemplates show form should render form with a valid response 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/new-invite.css\\"/><title>Veritable - New Connection</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon \\" style=\\"--background-image: url(&#34;/public/images/category.svg&#34;)\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"--background-image: url(&#34;/public/images/connection.svg&#34;)\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/folder.svg&#34;)\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/notification.svg&#34;)\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/setting.svg&#34;)\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">New Connection - PIN Verification</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><span class=\\"url-separator\\">/</span><a title=\\"Pin Submission\\" href=\\"/connection/new/XXXXXXXX/pin-submission\\">Pin Submission</a></div></div><div class=\\"content main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><div class=\\"card-body\\"><form id=\\"new-connection-form\\" hx-post=\\"/connection/new/pin-submission\\" hx-select=\\"#new-connection-form > *\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.5\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 2</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"connectionId\\" value=\\"XXXXXXXX\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" class=\\"new-connection-input-field\\" placeholder=\\"Code\\" required value=\\"\\" type=\\"text\\" pattern=\\"^[1-9][0-9]{5}$\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button   \\" href=\\"/connection\\"><span class=\\"button-text \\">Fill In Later</span></a><button class=\\"button   button-filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span class=\\"button-text \\">Continue</span></button></div></form></div></div></main></body></html>"`;

exports[`NewInviteTemplates show form should render form with a valid response 2`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx.org/ext/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/new-invite.css\\"/><title>Veritable - New Connection</title></head><body class=\\"flex-page\\" hx-ext=\\"json-enc\\"><nav class=\\"flex-page side-bar\\"><img class=\\"side-bar logo-container\\" src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" class=\\"side-bar icon \\" style=\\"--background-image: url(&#34;/public/images/category.svg&#34;)\\"></a><a title=\\"connections\\" href=\\"/connection\\" class=\\"side-bar icon active\\" style=\\"--background-image: url(&#34;/public/images/connection.svg&#34;)\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/folder.svg&#34;)\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/notification.svg&#34;)\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"side-bar icon disabled \\" style=\\"--background-image: url(&#34;/public/images/setting.svg&#34;)\\"></a></nav><main class=\\"flex-page content\\"><div class=\\"content header\\"><h1 class=\\"header heading\\">New Connection - PIN Verification</h1><div class=\\"header nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><span class=\\"url-separator\\">/</span><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><span class=\\"url-separator\\">/</span><a title=\\"Pin Submission\\" href=\\"/connection/new/00000001/pin-submission\\">Pin Submission</a></div></div><div class=\\"content main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><div class=\\"card-body\\"><form id=\\"new-connection-form\\" hx-post=\\"/connection/new/pin-submission\\" hx-select=\\"#new-connection-form > *\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.5\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 2</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"connectionId\\" value=\\"00000001\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" class=\\"new-connection-input-field\\" placeholder=\\"Code\\" required value=\\"123456\\" type=\\"text\\" pattern=\\"^[1-9][0-9]{5}$\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button   \\" href=\\"/connection\\"><span class=\\"button-text \\">Fill In Later</span></a><button class=\\"button   button-filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span class=\\"button-text \\">Continue</span></button></div></form></div></div></main></body></html>"`;

exports[`PinSubmissionTemplates show form should render form as a continuation of from invite 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/connection.css\\"/><title>Veritable - New Connection</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>New Connection - PIN Verification</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><a title=\\"Pin Submission\\" href=\\"/connection/CONNECTION_ID/pin-submission\\">Pin Submission</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><form id=\\"new-connection-form\\" hx-post=\\"/connection/CONNECTION_ID/pin-submission\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.6666666666666666\\"><div id=\\"new-connection-progress-text\\"><span>Step 2</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"stepCount\\" value=\\"3\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" placeholder=\\"Code\\" required value=\\"\\" type=\\"text\\" pattern=\\"^[0-9]{6}$\\" oninput=\\"this.reportValidity()\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Fill In Later</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span>Continue</span></button></div></form></div></main></body></html>"`;

exports[`PinSubmissionTemplates show form should render form as a stand alone flow 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/connection.css\\"/><title>Veritable - New Connection</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>New Connection - PIN Verification</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><a title=\\"Pin Submission\\" href=\\"/connection/CONNECTION_ID/pin-submission\\">Pin Submission</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><form id=\\"new-connection-form\\" hx-post=\\"/connection/CONNECTION_ID/pin-submission\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.5\\"><div id=\\"new-connection-progress-text\\"><span>Step 1</span><span> of 2</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"stepCount\\" value=\\"2\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" placeholder=\\"Code\\" required value=\\"\\" type=\\"text\\" pattern=\\"^[0-9]{6}$\\" oninput=\\"this.reportValidity()\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Fill In Later</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span>Continue</span></button></div></form></div></main></body></html>"`;

exports[`PinSubmissionTemplates show form should render form with PIN 1`] = `"<!DOCTYPE html><html lang=\\"en\\"><head><script src=\\"/lib/htmx.org/htmx.min.js\\"></script><script src=\\"/lib/htmx-ext-json-enc/json-enc.js\\"></script><script src=\\"/public/scripts/auth-redirect.js\\"></script><link rel=\\"icon\\" type=\\"image/ico\\" sizes=\\"48x48\\" href=\\"/public/images/favicon.ico\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/main.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/shared.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/mpa.css\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/public/styles/connection.css\\"/><title>Veritable - New Connection</title></head><body hx-ext=\\"json-enc\\"><nav id=\\"side-bar\\"><img src=\\"/public/images/logo-square.svg\\"/><a title=\\"categories\\" href=\\"/\\" style=\\"--background-image:url('/public/images/category.svg');\\"></a><a title=\\"connections\\" href=\\"/connection\\" data-active style=\\"--background-image:url('/public/images/connection.svg');\\"></a><a title=\\"storage\\" href=\\"#storage\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/folder.svg');\\"></a><a title=\\"notifications\\" href=\\"#notification\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/notification.svg');\\"></a><a title=\\"settings\\" href=\\"#settings\\" class=\\"disabled\\" style=\\"--background-image:url('/public/images/setting.svg');\\"></a></nav><main><div id=\\"content-header\\"><h1>New Connection - PIN Verification</h1><div id=\\"content-header-nav\\"><a title=\\"home\\" class=\\"nav icon\\" href=\\"/\\"></a><a title=\\"Connections\\" href=\\"/connection\\">Connections</a><a title=\\"Pin Submission\\" href=\\"/connection/CONNECTION_ID/pin-submission\\">Pin Submission</a></div></div><div id=\\"content-main\\"><div class=\\"connections header\\"><span>PIN Code submission</span></div><form id=\\"new-connection-form\\" hx-post=\\"/connection/CONNECTION_ID/pin-submission\\" hx-swap=\\"outerHTML\\" hx-select=\\"#new-connection-form\\"><div id=\\"new-connection-progress\\" style=\\"--progress-percent: 0.6666666666666666\\"><div id=\\"new-connection-progress-text\\"><span>Step 2</span><span> of 3</span></div><div id=\\"new-connection-progress-bar\\"></div></div><div class=\\"accented-container\\"><p>Please enter the verification code from the physical letter</p><input name=\\"stepCount\\" value=\\"3\\" type=\\"hidden\\"/><input id=\\"new-connection-invite-input-pin\\" name=\\"pin\\" placeholder=\\"Code\\" required value=\\"123456\\" type=\\"text\\" pattern=\\"^[0-9]{6}$\\" oninput=\\"this.reportValidity()\\" minlength=\\"6\\" maxlength=\\"6\\"/></div><div id=\\"new-connection-actions\\"><a class=\\"button \\" href=\\"/connection\\" data-variant=\\"outlined\\"><span>Fill In Later</span></a><button class=\\"button  \\" data-variant=\\"filled\\" type=\\"submit\\" name=\\"action\\" value=\\"submitPinCode\\"><span>Continue</span></button></div></form></div></main></body></html>"`;
